words/headers.asm

```diff
@@ -0,0 +1,629 @@
+#def_nt quit
+#def_nt begin,          CO+IM
+#def_nt again,          CO+IM+UF
+#def_nt state
+#def_nt evaluate,       UF
+#def_nt base
+#def_nt digit_question, UF, "digit?"
+#def_nt number,         UF
+#def_nt to_number,      UF, ">number"
+#def_nt hex
+#def_nt decimal
+#def_nt count,          UF
+#def_nt m_star,         UF, "m*"
+#def_nt um_star,        UF, "um*"
+#def_nt star,           UF, "*"
+#def_nt um_slash_mod,   UF, "um/mod"
+#def_nt sm_slash_rem,   UF, "sm/rem"
+#def_nt fm_slash_mod,   UF, "fm/mod"
+#def_nt slash,          UF, "/"
+#def_nt slash_mod,      UF, "/mod"
+#def_nt mod,            UF
+#def_nt star_slash_mod, UF, "*/mod"
+#def_nt star_slash,     UF, "*/"
+#def_nt backslash,      IM, "\"
+#def_nt move,           NN+UF
+#def_nt cmove_up,       UF, "cmove>"
+#def_nt cmove,          UF
+#def_nt pad
+#def_nt cleave,         UF
+#def_nt hexstore,       UF
+#def_nt within,         UF
+#def_nt to_in,          , ">in"
+#def_nt less_number_sign, , "<#"
+#def_nt number_sign,    UF, "#"
+#def_nt number_sign_s,  UF, "#s"
+#def_nt number_sign_greater, UF, "#>"
+#def_nt hold,           UF
+#def_nt sign,           UF
+#def_nt output
+#def_nt input
+#def_nt havekey
+#def_nt cr
+#def_nt page
+#def_nt at_xy,          UF, "at-xy"
+#def_nt marker,         IM
+#def_nt words
+#def_nt wordsize,       UF
+#def_nt aligned                           ; same code as ALIGN
+#def_nt align
+#def_nt bell
+#def_nt dump,           UF
+#def_nt dot_s,          , ".s"
+
+.if "disassembler" in TALI_OPTIONAL_WORDS
+#def_nt disasm,         UF
+.endif
+
+#def_nt compare,        UF
+#def_nt search,         UF+NN
+
+.if "environment?" in TALI_OPTIONAL_WORDS
+#def_nt environment_q,  UF, "environment?"
+.endif
+
+#def_nt find,           UF
+#def_nt word,           UF
+#def_nt paren,          IM, "("
+#def_nt dot_paren,      IM, ".("
+#def_nt if,             IM+CO+NN
+#def_nt then,           UF+IM+CO+NN
+#def_nt else,           UF+IM+CO+NN
+#def_nt repeat,         UF+IM+CO+NN
+#def_nt until,          UF+IM+CO+NN
+#def_nt while,          UF+IM+CO+NN
+#def_nt case,           IM+CO+NN        ; shares code with ZERO
+#def_nt of,             IM+CO+NN
+#def_nt endof,          UF+IM+CO+NN
+#def_nt endcase,        UF+IM+CO+NN
+#def_nt defer_fetch,    UF, "defer@"
+#def_nt defer_store,    UF, "defer!"
+#def_nt is,             IM
+#def_nt action_of,      IM, "action-of"
+#def_nt useraddr,       , "useraddr"
+#def_nt buffer_colon,   UF, "buffer:"
+
+.if "block" in TALI_OPTIONAL_WORDS
+#def_nt buffstatus
+#def_nt buffblocknum
+#def_nt blkbuffer
+#def_nt scr,            NN
+#def_nt blk,            NN
+;TODO these mention HC but don't have the flag?  or is the HC wrong in block_read* ?
+#def_nt block_write,    UF+NN, "block-write"    ; Deferred words need the HC (Code Field) flag.
+#def_nt block_write_vector, NN, "block-write-vector" ; Deferred words need the HC (Code Field) flag.
+#def_nt block_read,     UF+HC+NN, "block-read"  ; Deferred words need the HC (Code Field) flag.
```

Comment:
The block_read and block_read_vector words include the HC flag, and the block_write and block_write_vector mention the flag but don't add it.  Should they all have it?  Or is it mistaken and none of them should have it (that's 
 my guess but haven't explored that code much)?